### PR TITLE
Re arrange CNF-Tests related files to a single folder

### PR DESCRIPTION
**Pull request by fedepaol**
This PR moves all the cnf-tests related files under the `cnf-tests` folder.
It also alignes all the paths, the dockerfiles, and the package paths.

It was split in 3 different commits to make the review process easier.

### CODE REVIEW

File: cnf-tests/hack/compare-gen-md.sh [hunk 1]
```diff
@@ -4,8 +4,9 @@
 
 MDFILE="TESTLIST.md"
 
-mv cnf-tests/${MDFILE} _cache/${MDFILE}.old
+mv ${MDFILE} _cache/${MDFILE}.old
 
-. $(dirname "$0")/generate-cnf-docs.sh
+ls
```
Comment (oribon) [inline cnf-tests/hack/compare-gen-md.sh:9]: is this doing anything?

File: cnf-tests/hack/compare-gen-md.sh [hunk 2]
```diff
@@ -4,8 +4,9 @@
 
 MDFILE="TESTLIST.md"
 
-mv cnf-tests/${MDFILE} _cache/${MDFILE}.old
+mv ${MDFILE} _cache/${MDFILE}.old
```
Comment (oribon) [inline cnf-tests/hack/compare-gen-md.sh:7]: this fails if cnf-tests/_cache was not created yet
should we add mkdir -p _cache before?

File: cnf-tests/hack/compare-gen-md.sh [hunk 3]
```diff
@@ -4,8 +4,9 @@
...
 
-. $(dirname "$0")/generate-cnf-docs.sh
+ls
+hack/generate-cnf-docs.sh
 
-diff cnf-tests/${MDFILE} _cache/${MDFILE}.old -q || { echo "Docs should be regenerated and updated upstream. You can use hack/fill-empty-docs.sh and hack/generate-cnf-docs.sh"; exit 1; }
+diff ${MDFILE} _cache/${MDFILE}.old -q || { echo "Docs should be regenerated and updated upstream. You can use hack/fill-empty-docs.sh and hack/generate-cnf-docs.sh"; exit 1; }
```
Comment (oribon) [inline cnf-tests/hack/compare-gen-md.sh:12]: this should be updated to cnf-tests/hack/...

**Review by oribon — COMMENTED** [review]
:-)

File: cnf-tests/hack/compare-gen-md.sh (hunk 1 above)
Comment (fedepaol) [reply cnf-tests/hack/compare-gen-md.sh:9]: nope, that was for debugging purpouses, thanks for spotting it!

File: cnf-tests/hack/compare-gen-md.sh (hunk 3 above)
Comment (fedepaol) [reply cnf-tests/hack/compare-gen-md.sh]: done

File: cnf-tests/hack/compare-gen-md.sh (hunk 2 above)
Comment (fedepaol) [reply cnf-tests/hack/compare-gen-md.sh:7]: Right, not sure why I did not catch it.
Probably the path was created automatically by ginkgo or something.
Added it to the local common.sh file plus added the folder to gitignore

**Pull request merged**